### PR TITLE
fix: Completion array anyOf

### PR DESCRIPTION
**Pull request by p-spacek**
When object is `array anyOf` completion doesn't suggest `- (array item)`
```
test_array_anyOf_2objects:
  - obj1:
      name: 5  
  # not ok when type is anyOf

```
### What does this PR do?
This PR implement array completion on array anyOf type
image

### What issues does this PR fix or reference?
Partly: https://github.com/p-spacek/yaml-language-server/issues/5

### Is it tested? How?
unit test

### CODE REVIEW

File: test/fixtures/testArrayCompletionSchema.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            }
+          },
+          {
+            "type": "null"
+          }
+        ]
+      }
+    },
+    "type": "object"
+  }
```
Comment (evidolob) [inline test/fixtures/testArrayCompletionSchema.json:85]: Please add new line at the end of file

File: test/autoCompletion.test.ts [hunk 2]
```diff
@@ -1710,5 +1710,109 @@ suite('Auto Completion Tests', () => {
...
+      // it('Simple array object completion without "-" without any item', async () => {
+      //   const schema = require(path.join(__dirname, './fixtures/testArrayCompletionSchema.json'));
+      //   languageService.addSchema(SCHEMA_ID, schema);
+      //   const content = 'test_simpleArrayObject:\n  ';
+      //   const completion = parseSetup(content, content.length);
+      //   completion.then(function (result) {
+      //     assert.equal(result.items.length, 1);
+      //     assert.equal(result.items[0]?.label, '- (array item)');
+      //   });
+      // });
```
Comment (evidolob) [inline test/autoCompletion.test.ts:1724]: This file contains a lot of commented lines, can you delete them?

Comment (joshuawilson) [reply test/autoCompletion.test.ts:1724]: or if it is a failing test, get it to pass?

Comment (p-spacek) [reply test/autoCompletion.test.ts]: it fails for now, it should be fixed in the future

```
test_simpleArrayObject:
  #not ok when there is no '-' symbol (should suggest '- (array item)')
```

I removed them. Doesn't make sense to be there...
thanks guys

**Review by JPinkney — APPROVED** [review]
Played around with it a bit and it LGTM. I did find a different but unrelated issue to this PR though when testing
testyaml

obj1 and obj2 when autocompleting don't seem to indent correctly

**Review by evidolob — APPROVED** [review]

**Pull request merged**